### DISCUSSION TITLE
Barcode Scanning Plugin Not Reading Brazilian 44-Digit ITF Barcode Correctly

### DISCUSSION THREAD

**Discussion by jacksonpietro** [capawesome-team/capacitor-mlkit#147] [Q&A]
<p>Hello,</p>
<p>I&apos;ve encountered an issue with the Barcode Scanning plugin on the Android platform, where it doesn&apos;t accurately read the complete 44-digit ITF (Interleaved 2 of 5) barcodes commonly used in Brazil.</p>
<p><strong><strong>Issue Details:</strong></strong></p>
<ul>
    <li value="1"><strong><strong>Plugin Name:</strong></strong> Barcode Scanning</li>
    <li value="2"><strong><strong>Platform:</strong></strong> Android</li>
    <li value="3"><strong><strong>Plugin Version:</strong></strong> 5.4.0</li>
    <li value="4"><strong><strong>Problem:</strong></strong> The plugin fails to read the entire 44-digit ITF barcodes correctly.</li>
    <li value="5"><strong><strong>Context:</strong></strong> This issue pertains specifically to the 44-digit ITF barcodes used in Brazilian documents such as boletos banc&aacute;rios (banking invoices).</li>
    <li value="6"><strong><strong>Expected Behavior:</strong></strong> The plugin should be able to accurately read and decode all 44 digits of the ITF barcodes.</li>
    <li value="7"><strong><strong>Observed Behavior:</strong></strong> The plugin either fails to read some digits or misinterprets parts of the barcode, resulting in incorrect data.</li>
    <li value="8"><strong><strong>Impact:</strong></strong> This issue affects the usability and reliability of the Barcode Scanning plugin for users dealing with Brazilian documents containing 44-digit ITF barcodes.</li>
</ul>
<p><strong><strong>Request for Assistance:</strong></strong></p>
<p>I kindly request assistance from the development team to address this issue. It&apos;s crucial for the Barcode Scanning plugin on Android to reliably handle the complete 44-digit ITF barcodes used in Brazil. Could the team investigate and provide guidance on resolving this problem?</p>
<p><strong><strong>Additional Information:</strong></strong></p>
<ul>
    <li value="1"><strong><strong>Steps to Reproduce:</strong></strong> The issue occurs consistently when scanning Brazilian boletos banc&aacute;rios containing 44-digit ITF barcodes.</li>
</ul>
<p>Thank you for your attention to this matter.</p>
<p>Best regards, Jackson Pietro</p>

**Comment by robingenz** [comment]
This sounds like a problem in the underlying ML Kit library. Please create an issue in https://github.com/googlesamples/mlkit.
